webcomponents/core/src/components/deck/deckdeckgo-deck/deckdeckgo-deck.tsx

```diff
@@ -1140,6 +1143,30 @@ export class DeckdeckgoDeck {
 
   /* END: Reveal */
 
+  /* BEGIN: AutoSlide */
+
+  @Watch('autoSlide')
+  onAutoSlide() {
+    let idleMouseTimer;
+
+    setTimeout(() => (idleMouseTimer = this.idleMouseTimer), 2000);
+
+    if (this.autoSlide) {
+      this.slideLoopInterval = setInterval(() => {
```

Comment:
Both `slideNext` and `slideTo` are async. Can you set the function as such and add the `await` for both functions?